src/ray/core_worker/core_worker.cc

```diff
@@ -182,14 +215,36 @@ CoreWorkerProcess::CoreWorkerProcess(const CoreWorkerOptions &options)
   // NOTE(lingxuan.zlx): We assume RayConfig is initialized before it's used.
   // RayConfig is generated in Java_io_ray_runtime_RayNativeRuntime_nativeInitialize
   // for java worker or in constructor of CoreWorker for python worker.
-  stats::Init(global_tags, options_.metrics_agent_port);
+  stats::Init(global_tags, [this](const stats::GetAgentAddressCallback &callback) {
+    worker_map_mutex_.Lock();
+    auto current_node_id = current_node_id_;
+    worker_map_mutex_.Unlock();
+    if (current_node_id.IsNil()) {
+      callback(ray::Status::Invalid("No core worker created."), std::string());
+      return;
+    }
+    GetAgentAddress(gcs_client_, current_node_id, callback);
+  });
 
   // Initialize event framework.
   if (RayConfig::instance().event_log_reporter_enabled() && !options_.log_dir.empty()) {
     RayEventInit(ray::rpc::Event_SourceType::Event_SourceType_CORE_WORKER,
                  std::unordered_map<std::string, std::string>(), options_.log_dir);
   }
 
+  io_thread_ = std::thread([&] {
```

Comment:
we are also adding a new io_thread_ to core worker here. Is this required? maybe there are other io_service_ or context that can be reused? @rkooo567

Reply:
Currently, each core worker binds to a thread and owns one io service. It's more complex to share the core worker io service to the stats (global in a  process).

Reply:
@kfstorm Do you have a better solution?

Reply:
I think creating a new thread in `CoreWorkerProcess` for the process-scope event loop is reasonable (we already did this internally). However, I do have some concerns about the additional GCS client. Maybe we can open-source our internal approach of sharing a single GCS client among multiple `CoreWorker` instances and the `CoreWorkerProcess` instance.

@simon-mo @fyrestone If you both agree with this, I can create another PR to open-source it.

Reply:
@fyrestone Thanks. It sounds good. sharing the GCS client will reduce the connections from core worker to GCS.